packages/firestore/src/core/sync_engine_impl.ts

```diff
@@ -331,6 +328,10 @@ export async function syncEngineListen(
       status === 'current',
       targetData.resumeToken
     );
+
+    if (syncEngineImpl.isPrimaryClient) {
```

Comment:
Is it actually necessary to move this "chunk" of code around? I had originally thought that the return value of `addLocalQueryTarget()` above could have been used somehow to get the `expectedCount`, avoiding having to call `getRemoteKeysForTarget()` in `remoteStoreListen()`. But that appears to not be the case, in which case it makes sense to still send off query to the backend before computing the local snapshot.

Reply:
After discussion, it is indeed required to call `remoteStoreListen()` _after_ `initializeViewAndComputeSnapshot()` because the latter initializes some data in the sync engine that `getRemoteKeysForTarget()` uses, namely `queriesByTarget` and `queryViewsByQuery`.